lib/actions-oclif/envs.ts

```diff
@@ -202,6 +266,22 @@ export default class EnvsCmd extends Command {
 			if (!options.config) {
 				fields.push(options.json ? 'serviceName' : 'serviceName => SERVICE');
 			}
+		} else {
+			if (options.all) {
+				// Replace undefined app names with 'N/A' or null
+				varArray = _.map(varArray, (i: EnvironmentVariableInfo) => {
+					i.appName = i.appName || (options.json ? null : 'N/A');
+					return i;
+				});
+
+				fields.push(options.json ? 'appName' : 'appName => APPLICATION');
+				if (options.device) {
+					fields.push(options.json ? 'deviceUUID' : 'deviceUUID => DEVICE');
+				}
+				if (!options.config) {
+					fields.push(options.json ? 'serviceName' : 'serviceName => SERVICE');
+				}
+			}
```

Comment:
Wouldn't it be simpler and much less code duplication to have the following? Early on in the `run()` method:
```
options.all = options.all || isV12();
```
Then all code that currently refers to `options.all` can remain unchanged. After v12 is released, we can go around deleting some redundant conditionals.